\documentclass[a4paper,12pt]{article}
\usepackage{nameref}
\usepackage{grffile}
\usepackage{graphicx}
\usepackage[strings]{underscore}
\usepackage{verbatim}
\usepackage{wrapfig}
\usepackage{lastpage}

\begin{comment}
#
# =====================================================
# $Id: QCReportTemplate.tex 11716 2012-04-20 11:55:31Z mdijkstra $
# $URL: http://www.molgenis.org/svn/molgenis_apps/trunk/modules/compute/protocols/QCReportTemplate.tex $
# $LastChangedDate: 2012-04-20 13:55:31 +0200 (Fri, 20 Apr 2012) $
# $LastChangedRevision: 11716 $
# $LastChangedBy: mdijkstra $
# =====================================================
#
\end{comment}

\newenvironment{narrow}[2]{
  \begin{list}{}{
    \setlength{\leftmargin}{#1}
    \setlength{\rightmargin}{#2}
    \setlength{\listparindent}{\parindent}
    \setlength{\itemindent}{\parindent}
    \setlength{\parsep}{\parskip}
  }
  \item[]
}{\end{list}}


\begin{document}
\thispagestyle{empty}
\vspace{40mm}


\clearpage
\section*{README}

This zip file contains the following files and directories:
\subsection*{SNPs}
*.snps.final.vcf = All SNP calls in VCF format. (eg. to be used for Cartagenia etc.)

\noindent*.snps.final.vcf.table = All SNP calls in tab-delimited format.

\subsection*{QC}
*_fastqc.zip = Information about the quality of the raw sequence reads. Please read the FastQC manual page for detailed information.

\#\#\#OPTIONAL\#\#\#

\noindent*.genotypeArray.updated.header.vcf = Sample genotype calls in VCF format.


\subsection*{QC/statistics}
*metric files = Metrics used to compile the QC report. For information/documentation about all metrics not explained in the QC report we refer to the Picard documentation.


\subsection*{}


${project}.csv = This file contains all the information for the project. If samples were pooled information about the used barcodes etc. can be found here.

\noindent ${project}_QCReport.pdf = QC report containing all important QC metrics per sample.


\subsection*{}


\subsection*{Additional files}

*.merged.bam = The BAM file containing all (un)aligned reads after removing duplicates and quality score recalibration. 
This is the final alignment on which SNP calling was executed.

\noindent*.merged.bam.bai = The index file of the *.merged.bam.

\noindent*.fq.gz = gzipped FASTQ file containing all reads, if multiple samples were pooled these files contain the demultiplexed reads per flowcell, lane, barcode combination. The exact combination for each sample can be found in the ${project}.csv file.

\noindent*.fq.md5 = the md5sum for the corresponding FASTQ file.


\subsection*{}


\section*{Documentation}
FastQC: http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/
Picard: http://picard.sourceforge.net/picard-metric-definitions.shtml

\end{document}