
#NGS QC report
##Next Generation Sequencing report
##Genome Analysis Facility (GAF), Genomics Coordination Centre (GCC) University Medical Centre Groningen


.|.
---|---
**By** | GAF, GCC, UMCG
**On** | `r strsplit(as.character(Sys.time()), " ")[[1]][1] # get date`
**Project** | `r Sys.getenv("project")`
**Customer**|
**Principal investigator**| `r Sys.getenv("contact")`
**Generated by** | MOLGENIS Compute
**Contact**|
**Name**|Cleo C. van Diemen
**E-mail**|c.c.van.diemen@umcg.nl


##Introduction
This report describes a series of statistics about your sequencing data. Together with this report you’ll receive a SNP-list. 
If you, in addition, also want the raw data, then please notify us via e-mail. In any case we’ll delete the raw data, three months 
after `r strsplit(as.character(Sys.time()), " ")[[1]][1]`.


##Statistics
```{r echo=FALSE,results='asis'}
	cat(csv.to.markdown.table.CUSTOM.qc( qcStatisticsCsv ))
```

##Name of the pipeline used in this project
```{r engine='bash', echo=F}
        echo ${dnaVersion}
```


##Name of the bait set(s) used in the hybrid selection for this project:
```{r engine='bash', echo=F}
	cat ${qcBaitSet}
```


##Description statistics table

1.	External sample name
2.	Number of bases in the reference genome used for alignment 
3.	Number of bases which have one or more baits on top of them 
4.	Unique number of target bases in the experiment where target is usually exons etc
5.	Total number of reads including all PF and non-PF reads
6.	Number of bases in the PF aligned reads that are aligned to the reference genome. Accounts for clipping and gaps (=indels?) 
7.	Mean coverage of all baits
8.	Mean coverage of targets with coverage ≥ 2
9.	Fraction bp near bait (+/- 250 bp)
10.	Capture specificity; 1 - fraction bases that is off bait
11.	Percentage of ALL target bases with coverage ≥ 2
12.	Percentage of ALL target bases with coverage ≥ 10
13.	Percentage of ALL target bases with coverage ≥ 20
14.	Percentage of ALL target bases with coverage ≥ 30
15.	Fraction of aligned, deduplicated, on-target bases out of the PF bases available
16.	Mean read length of the set of reads examined
17.	Number of PF reads aligned to the positive strand of the genome divided by the number of PF reads aligned to the genome
18.	Median of the insert sizes, where insert refers to the base pairs between the adapters.
19.	Mean of the insert sizes, where insert refers to the base pairs be- tween the adapters.
20.	Standard deviation of the insert sizes, where insert refers to the base pairs between the adapters.
21.	Fraction of SNPs that are concordant between array and NGS


##Duplication rates
```{r echo=FALSE,results='asis'}
        cat(csv.to.markdown.table( qcDedupMetricsOut ))
```

##Insert size distribution
The following figures show the insert size distribution per sample. Insert refers to the base pairs that are ligated between the adapters.
```{r echo=FALSE,results='asis'}
   for (i in 1:length(sampleInsertSizePDF)) {
        showFigure(sampleInsertSizePDF[i], paste("Coverage plot of", externalSampleID[i]))
    }
```
##In silico concordance
To auto-validate our in silico analysis pipeline, we spike each sample with simulated reads that originate from the PhiX genome. Those simulated reads contain some 
de novo mutations compared to the PhiX reference genome (http://www.ncbi.nlm.nih.gov/nuccore/9626372?report=fasta), which should be detected by our pipeline:
```{r engine='bash', echo=F}
        cat ${inSilicoConcordanceFile}
```
